eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/FieldValue/Handler.php

```diff
@@ -147,4 +159,75 @@ protected function lowerCase(string $string): string
     {
         return $this->transformationProcessor->transformByGroup($string, 'lowercase');
     }
+
+    /**
+     * @param scalar|array<scalar> $value
+     *
+     * @return scalar|array<scalar>
+     */
+    private function prepareParameter($value)
+    {
+        if (is_string($value)) {
+            return $this->lowerCase($value);
+        } elseif (is_array($value)) {
+            return array_map([$this, 'prepareParameter'], $value);
+        }
+
+        return $value;
+    }
+
+    private function createNamedParameter(QueryBuilder $outerQuery, string $column, $value): ?string
+    {
+        switch ($column) {
+            case 'sort_key_string':
+                $parameterValue = $this->prepareParameter($value);
+                $parameterType = ParameterType::STRING;
+                break;
+            case 'sort_key_integer':
+                $parameterValue = (int)$value;
+                $parameterType = ParameterType::INTEGER;
+                break;
+            default:
+                $parameterValue = $value;
+                $parameterType = null;
+        }
+
+        return $outerQuery->createNamedParameter(
+            $parameterValue,
+            $parameterType
+        );
+    }
+
+    private function isOperatorUnary(string $operator): bool
+    {
+        return !in_array($operator, [
+            Criterion\Operator::IN,
+            Criterion\Operator::BETWEEN,
+        ], true);
+    }
+
+    /**
+     * @param array<int, scalar> $values
+     *
+     * @throws \eZ\Publish\API\Exception\InvalidArgumentException If value contain unhandled or more than one parameter type
```

Comment:
This class does not exist:
```suggestion
     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException If value contain unhandled or more than one parameter type
```